test/services/SubmissionServiceSpec.scala

```diff
@@ -41,156 +43,286 @@ class SubmissionServiceSpec
+        mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+
+      }
+    }
+
+    "the user has no Nino and enable(EnableNoNinoJourney)" should {
+      s"not register but create a trn"  in  {
+        val matchingResult = Random.nextBoolean()
+
+        enable(EnableNoNinoJourney)
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoNinoNoSautr)))
+        mockMatchSoleTraderDetailsNoNino(testJourneyId, testIndividualDetailsNoNinoNoSautr)(Future.successful(Right(matchingResult)))
+        mockCreateTrn(testJourneyId)(Future.successful(testTrn))
+        mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+
+      }
+    }
+
+    "the user has no Nino and disable(EnableNoNinoJourney)" should {
+      s"not register and not create a trn"  in  {
+
+        val matchingResult = Random.nextBoolean()
+
+        disable(EnableNoNinoJourney)
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoNinoNoSautr)))
+        mockMatchSoleTraderDetails(testJourneyId, testIndividualDetailsNoNinoNoSautr, journeyConfigWithoutBV)(Future.successful(Right(matchingResult)))
+        mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+
+      }
+    }
+
+    "there is a matching error" should {
+      s"return a SoleTraderDetailsMismatch"  in  {
+
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetails)))
+        mockMatchSoleTraderDetails(testJourneyId, testIndividualDetails, journeyConfigWithoutBV)(Future.successful(Left(DetailsMismatch)))
+        mockStoreIdentifiersMatch(testJourneyId, identifiersMatch = true)(Future.successful(SuccessfullyStored))
```

Comment:
This should be false

Reply:
As before mockStoreIdentifiersMatch is never called in the code